theano/tensor/nnet/abstract_conv.py

```diff
@@ -430,6 +439,32 @@ def do_constant_folding(self, node):
         # This may change in the future.
         return False
 
+    def conv2d(self, img, kern, mode="valid"):
+        """
+        Basic slow python implementatation for DebugMode
+        """
+
+        if not imported_scipy_signal:
+            raise NotImplementedError(
+                "AbstractConv perform requires the python package"
+                " for scipy.signal to be installed.")
+        if not (mode in ('valid', 'full')):
+            raise ValueError(
+                'invalid mode {}, which must be either '
+                '"valid" or "full"'.format(mode))
+
+        out_shape = get_conv_output_shape(img.shape, kern.shape, mode, [1, 1])
+        out = numpy.zeros(out_shape, dtype=img.dtype)
+        val = _valfrommode(mode)
+        bval = _bvalfromboundary('fill')
+        for b in xrange(img.shape[0]):
+            for n in xrange(kern.shape[0]):
+                for im0 in xrange(img.shape[1]):
```

Comment:
`xrange` is not defined in python 3, so Travis is failing.
You can import it from `six.moves` to be compatible with both.